File: MošPosudit.Services/DataBase/Data/SystemLog.cs
using System.ComponentModel.DataAnnotations;
using MošPosudit.Model.Enums;

namespace MošPosudit.Services.DataBase.Data
{
    public class SystemLog
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public DateTime Timestamp { get; set; }

        [Required]
        public LogLevel LogLevel { get; set; }

        [Required]
        public LogAction Action { get; set; }

        [Required]
        [StringLength(100)]
        public string Entity { get; set; } // User, Tool, Rental, etc.

        public int? EntityId { get; set; } // ID of the affected entity

        [StringLength(500)]
        public string Message { get; set; }

        [StringLength(100)]
        public string Username { get; set; } // User who performed the action

        [StringLength(50)]
        public string IpAddress { get; set; }

        [StringLength(500)]
        public string AdditionalInfo { get; set; } // JSON string for additional data

        [StringLength(500)]
        public string StackTrace { get; set; } // For error logs
    }
} 

File: MosPosudit.UI/shared/lib/models/system_log.dart
class SystemLogModel {
  final int id;
  final DateTime timestamp;
  final String? logLevel;
  final String? action;
  final String? entity;
  final int? entityId;
  final String? message;
  final String? username;
  final String? ipAddress;
  final String? additionalInfo;
  final String? stackTrace;
  final String? details;

  SystemLogModel({
    required this.id,
    required this.timestamp,
    this.logLevel,
    this.action,
    this.entity,
    this.entityId,
    this.message,
    this.username,
    this.ipAddress,
    this.additionalInfo,
    this.stackTrace,
    this.details,
  });

  factory SystemLogModel.fromJson(Map<String, dynamic> json) => SystemLogModel(
        id: json['id'],
        timestamp: DateTime.parse(json['timestamp']),
        logLevel: json['logLevel'],
        action: json['action'],
        entity: json['entity'],
        entityId: json['entityId'],
        message: json['message'],
        username: json['username'],
        ipAddress: json['ipAddress'],
        additionalInfo: json['additionalInfo'],
        stackTrace: json['stackTrace'],
        details: json['details'],
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'timestamp': timestamp.toIso8601String(),
        'logLevel': logLevel,
        'action': action,
        'entity': entity,
        'entityId': entityId,
        'message': message,
        'username': username,
        'ipAddress': ipAddress,
        'additionalInfo': additionalInfo,
        'stackTrace': stackTrace,
        'details': details,
      };
}

